gap/base/recognition.gi

```diff
@@ -708,6 +708,11 @@ InstallGlobalFunction( SLPforElementGeneric,
     local gg,n,rifac,riker,s,s1,s2,y,nr1,nr2;
     rifac := RIFac(ri);
     riker := RIKer(ri);   # note: might be fail
+
+    if HasValidateHomomInput(ri) and not ValidateHomomInput(ri)(g) then
+        return fail;
+    fi;
+
     gg := ImageElm(Homom(ri),g);
```

Comment:
There in total 5 places where we call `ImageElm`. I think for each of the 4 other places, we should either add a call to `ValidateHomomInput`, or else add a comment that explains why a call to `ValidateHomomInput` is not necessary.